### PR TITLE
[Ionut Demsa] iterator_task_1

### PR DESCRIPTION
**Pull request by Ionut95**

### CODE REVIEW

**Comment by Ionut95** [comment]
I didn't manage to include stream iterators. I managed to learn something more about them. In order to include them I thought of copying the stringstream in another one by using istream_iterator, but not sure how to complete this: **std::istream_iterator<std::string>(mystringstream), std::istream_iterator<std::string>()** and do I have to initialize another string with this?

File: iterators_task_1/student/include/impl.h [hunk 1]
```diff
@@ -10,4 +10,14 @@
  *
  * @warning YOU MUST USE STRING STREAMS AND STREAM ITERATORS!
  */
-std::string reverseWords(const std::string& input);
+std::string reverseWords(const std::string& input) {
+    std::stringstream myStream(input);
+    std::string token, reversedSentence;
+    while(myStream >> token) {
```
Comment (smay1613) [inline iterators_task_1/student/include/impl.h]: 1. Use std::copy
2. Or initialize container with pair of stream iterators (begin is pointing to the stream, second is default-initialized)
3. Use reverse iterators/prepend to the container

File: iterators_task_1/student/include/impl.h [hunk 2]
```diff
@@ -10,4 +10,14 @@
...
+        reversedSentence = " " + token + reversedSentence;
+    }
+    if (reversedSentence.empty()) {
+        return "";
+    }
+    return reversedSentence.substr(1);
```
Comment (smay1613) [inline iterators_task_1/student/include/impl.h]: pop?

**Comment by smay1613** [comment]
Try to use container as an temporary storage between in/out streams

File: iterators_task_1/student/include/impl.h [hunk 3]
```diff
@@ -11,13 +14,15 @@
...
+    std::stringstream stream(input);
+    std::vector<std::string> tokens;
+
+    std::copy(std::istream_iterator<std::string>(stream), std::istream_iterator<std::string>(), std::back_inserter(tokens));
+    if(tokens.empty()) {
         return "";
     }
-    return reversedSentence.substr(1);
+    std::string reversed_sentence;
+    for_each(tokens.rbegin(), tokens.rend(), [&reversed_sentence](const std::string& item){reversed_sentence += item + " ";});
```
Comment (smay1613) [inline iterators_task_1/student/include/impl.h]: avoid for each, use output stream iterators here

**Pull request closed without merging**